### PR TITLE
cli: download-files --verify each file as they are downloaded

### LINKED ISSUE #63: download-files: --verify each file as they are downloaded
Currently, the files are being verified at the end of a download:

```console
$ cernopendata-client download-files --recid 5500 --filter-range 1-3 --verify
==> Downloading file 1 of 3
  -> File: ./5500/BuildFile.xml
  -> Progress: 0/0 kiB (100%)
==> Downloading file 2 of 3
  -> File: ./5500/HiggsDemoAnalyzer.cc
  -> Progress: 81/81 kiB (100%)
==> Downloading file 3 of 3
  -> File: ./5500/List_indexfile.txt
  -> Progress: 1/1 kiB (100%)
==> Verifying downloaded files for record 5500.
==> Verifying file BuildFile.xml...
  -> expected size 305, found 305
  -> expected checksum adler32:ff63668a, found adler32:ff63668a
==> Verifying file HiggsDemoAnalyzer.cc...
  -> expected size 83761, found 83761
  -> expected checksum adler32:f205f068, found adler32:f205f068
==> Verifying file List_indexfile.txt...
  -> expected size 1669, found 1669
  -> expected checksum adler32:46a907fc, found adler32:46a907fc
==> Success!
```

It would be good to improve this so that each file is immediately verified after it is downloaded:
 
- first filter which files user wants to download
- then for each file to be downloaded, do:
   - download it
   - check its size and sha1
- print success

This is so that the user can spot troubles early, not only after downloading all the files.

_Originally posted by @tiborsimko in https://github.com/cernopendata/cernopendata-client/pull/58#discussion_r502380191_

### PR DESCRIPTION
**Pull request by ParthS007**
closes #63

### CODE REVIEW

File: cernopendata_client/cli.py [hunk 1]
```diff
@@ -284,14 +284,12 @@ def download_files(
...
-        )
-        file_info_local = get_file_info_local(recid)
-        click.secho("==> Verifying downloaded files for record {}.".format(recid))
-        verify_file_info(file_info_local, file_info_remote)
+        if verify:
+            file_info_remote = get_file_info_remote(
+                recid, filtered_files=[file_location]
+            )
+            file_info_local = get_file_info_local(recid)
+            verify_file_info(file_info_local, file_info_remote)
```
Comment (tiborsimko) [inline cernopendata_client/cli.py:292]: Looks good in general :+1: 

```console
$ cernopendata-client download-files --recid5500 --filter-range 1-4 --verify
==> Downloading file 1 of 4
  -> File: ./5500/BuildFile.xml
  -> Progress: 0/0 kiB (0%)
==> Verifying file BuildFile.xml...
  -> expected size 305, found 92
ERROR: File size does not match.
```

However I noticed that when I use `--server` option, the verification does not seem to be done:

```console
$ cernopendata-client download-files --recid 5500 --filter-range 1-4 --verify --server http://localhost:5000
==> Downloading file 1 of 4
  -> File: ./5500/BuildFile.xml
  -> Progress: 7/7 kiB (100%)
==> Downloading file 2 of 4
  -> File: ./5500/HiggsDemoAnalyzer.cc
  -> Progress: 7/7 kiB (100%)
==> Downloading file 3 of 4
  -> File: ./5500/List_indexfile.txt
  -> Progress: 7/7 kiB (100%)
==> Downloading file 4 of 4
  -> File: ./5500/M4Lnormdatall.cc
  -> Progress: 7/7 kiB (100%)
==> Success!
```

Can you please have a look?

Comment (ParthS007) [reply cernopendata_client/cli.py:292]: Looking into it 👍

Comment (tiborsimko) [reply cernopendata_client/cli.py:292]: This is OK:

```console
$ cernopendata-client download-files --recid 5500 --filter-range 1-4 --verify --server http://opendata.cern.ch
==> Downloading file 1 of 4
  -> File: ./5500/BuildFile.xml
  -> Progress: 0/0 kiB (100%)
==> Verifying file BuildFile.xml...
  -> expected size 305, found 305
  -> expected checksum adler32:ff63668a, found adler32:ff63668a
...
```

This is not:

```console
$ cernopendata-client download-files --recid 5500 --filter-range 1-4 --verify --server http://opendata-dev.cern.ch
==> Downloading file 1 of 4
  -> File: ./5500/BuildFile.xml
  -> Progress: 0/0 kiB (100%)
==> Downloading file 2 of 4
  -> File: ./5500/HiggsDemoAnalyzer.cc
  -> Progress: 81/81 kiB (100%)
...
```

Comment (ParthS007) [reply cernopendata_client/cli.py:292]: Done 👍

**Review by tiborsimko — APPROVED** [review]

**Pull request merged**